src/poller/utils.rs

```diff
@@ -232,7 +301,7 @@ pub async fn save_url_to_internet_archive_urls(
     from_table: &str,
     from_table_id: i32,
     pool: &PgPool) {
-    if should_insert_url_to_internet_archive_urls(url, pool).await.expect("Error: ") {
+    if let Ok(_)= should_insert_url_to_internet_archive_urls(url, pool).await {
```

Comment:
Sorry for the backseat review, but this line seems really suspicious in both logic and style. Here's a few things that are wrong IMO

- The condition is equivalent to checking if the return value of the function is `Ok` or `Err`. It doesn't check the returned boolean. You need to unwrap the result first to get the boolean. If you are actually checking whether the result "is ok", prefer using `Result::is_ok()` as it's more readable
- You are discarding the `Err` value silently. Usually you should explicitly deal with the `Err` value by either doing `.expect()`, `match` or just sending it to the return value with `?`. 
- If you don't care about the `Err` value, it's recommended to put a comment to tell why this is fine
- The formating isn't correct. IDK what policy there is for formating, but you can use `cargo fmt` to easily format your code

Reply:
Hi @RustyNova016 thanks a lot for the review, I forgot that we were returning `bool` wrapped Result from the function `should_insert_url_to_internet_archive_urls` and assumed it was fine to check if the result is `Ok`. 
Also regarding the Err, I think we should generalize this for all util functions that returns Result.
- Cooool, I think I should automate cargo fmt some how on each commit, that would help format it before pushing
Thanks a ton, it really helped a lot 😁

Reply:
Glad to help and finally contribute to an Metabrainz project. Might make some PR to fix some parts to make them more idiomatic. I suppose you're a quite new rustacean so it's understandable 

Although I did notice that your fix commit included the whole reformating of the app. Maybe do it a separate PR? Other reviewer may find it hard to review the PR now that it is filed with formatting changes.

As for auto reformating, I don't know how to do it outside of Rust Rover. But I do know how to use CI to refuse unformatted PRs.

Reply:
Exemple here
https://github.com/RustyNova016/listenbrainz-cli-tools/blob/master/.github%2Fworkflows%2Frust.yml#L116

Reply:
Thanks, will update the PR soon, will get rid of the latest commit and update with newer one with  changes only in `save_url_to_internet_archive_urls` function.
Regarding the CI, I haven't added any CI pipeline yet, but will have a look at the example and add a workflow as well. ✅